cppe/CMakeLists.txt

```diff
@@ -62,35 +44,69 @@ add_library(cppe ${PYTHON_SHARED}
   core/electric_fields.cc
   core/math.cc
   core/pe_energies.cc
-  ${PYTHON_INTERFACE_SOURCES} ${PELIB_INTERFACE_SOURCES}
 )
 
+add_library(cppe
+  ${CPPE_SOURCES}
+  ${PELIB_INTERFACE_SOURCES}
+)
+
+
 target_include_directories(cppe PUBLIC ${EIGEN3_INCLUDE_DIRS})
 
 if(${ENABLE_PYTHON_INTERFACE})
-    target_link_libraries(cppe PUBLIC pybind11::module)
-    set_target_properties(cppe PROPERTIES PREFIX "${PYTHON_MODULE_PREFIX}"
-                                             SUFFIX "${PYTHON_MODULE_EXTENSION}")
+    message(STATUS "Building Python interface for cppe.")
+
+    if(MSVC)
+      set(PYBIND11_CPP_STANDARD "/std:c++${CMAKE_CXX_STANDARD}")
+    else()
+      set(PYBIND11_CPP_STANDARD "-std=c++${CMAKE_CXX_STANDARD}")
+    endif()
+    find_package(pybind11 CONFIG REQUIRED)
+    message(STATUS "${Cyan}Using pybind11${ColourReset}: ${pybind11_INCLUDE_DIR} (version ${pybind11_VERSION} for Py${PYTHON_VERSION_STRING} and ${PYBIND11_CPP_STANDARD})")
+    message(STATUS "${Cyan}Using Python ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}${ColourReset}: ${PYTHON_EXECUTABLE}")
+
+    pybind11_add_module(pycppe SHARED NO_EXTRAS
+        ${CPPE_SOURCES}
+        python_iface/export_molecule.cc
+        python_iface/export_multipole.cc
+        python_iface/export_options.cc
+        python_iface/export_fields.cc
+        python_iface/export_utils.cc
+        python_iface/export_state.cc
+        python_iface/export_math.cc
+        python_iface/export_cppe.cc
+    )
+    target_include_directories(pycppe PUBLIC ${EIGEN3_INCLUDE_DIRS})
+    set_target_properties(pycppe PROPERTIES PREFIX "${PYTHON_MODULE_PREFIX}"
+                                            OUTPUT_NAME cppe
+                                            EXPORT_NAME cppe
+                                            SUFFIX "${PYTHON_MODULE_EXTENSION}")
 endif()
 
 set(PN "cppe")
 #### Installation ####
-option(INSTALL_DEVEL_HEADERS "Install devel headers" ON)
 if (INSTALL_DEVEL_HEADERS)
     install(DIRECTORY "core"
             DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PN}
             FILES_MATCHING PATTERN "*.hh")
     install(DIRECTORY "utils"
             DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PN}
             FILES_MATCHING PATTERN "*.hh")
-    install(FILES libcppe.hh
-            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PN})
 endif()
+install(FILES libcppe.hh
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PN})
+
 install(TARGETS cppe
         EXPORT "${PN}Targets"
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
+install(TARGETS pycppe
```

Comment:
One should check here if the Python interface is enabled, otherwise, the target will not exist and CMake will fail.